src/randomx.cpp

```diff
@@ -293,7 +298,10 @@ extern "C" {
 	void randomx_vm_set_cache(randomx_vm *machine, randomx_cache* cache) {
 		assert(machine != nullptr);
 		assert(cache != nullptr && cache->isInitialized());
-		machine->setCache(cache);
+		if (machine->cacheData != cache->cacheData) {
+			machine->setCache(cache);
+			machine->cacheData = cache->cacheData;
+		}
```

Comment:
You need to set `machine->cacheData` also on line 283 when the VM is created.

Reply:
set